src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightLambda.cs

```diff
@@ -33,6 +34,175 @@ internal LightLambda(LightDelegateCreator delegateCreator, IStrongBox[] closure)
+                        {
+                            foreach (ExceptionHandler catchHandler in handler.Handlers)
+                            {
+                                _handlerEnter.Add(catchHandler.HandlerStartIndex - 1 /* include EnterExceptionHandler instruction */, catchHandler.ToString());
+                                AddHandlerExit(catchHandler.HandlerEndIndex);
+
+                                ExceptionFilter filter = catchHandler.Filter;
+                                if (filter != null)
+                                {
+                                    _handlerEnter.Add(filter.StartIndex - 1 /* include EnterExceptionFilter instruction */, "filter");
+                                    AddHandlerExit(filter.EndIndex);
+                                }
+                            }
+                        }
+                    }
+
+                    var enterTryFault = instruction as EnterTryFaultInstruction;
+                    if (enterTryFault != null)
+                    {
+                        TryFaultHandler handler = enterTryFault.Handler;
+
+                        AddTryStart(handler.TryStartIndex);
+                        AddHandlerExit(handler.TryEndIndex + 1 /* include Goto instruction that acts as a "leave" */);
+
+                        _handlerEnter.Add(handler.FinallyStartIndex, "fault");
+                        AddHandlerExit(handler.FinallyEndIndex);
+                    }
+                }
+            }
+
+            private void AddTryStart(int index)
+            {
+                int count;
+                if (!_tryStart.TryGetValue(index, out count))
+                {
+                    _tryStart.Add(index, 1);
+                    return;
+                }
+
+                _tryStart[index] = count + 1;
+            }
+
+            private void AddHandlerExit(int index)
+            {
+                int count;
+                if (!_handlerExit.TryGetValue(index, out count))
+                {
+                    _handlerExit.Add(index, 1);
+                    return;
+                }
+
+                _handlerExit[index] = count + 1;
+            }
+
+            private void Indent()
+            {
+                _indent = new string(' ', _indent.Length + 2);
+            }
+
+            private void Dedent()
+            {
+                _indent = new string(' ', _indent.Length - 2);
+            }
+
+            public override string ToString()
+            {
+                var sb = new StringBuilder();
+
+                string name = _interpreter.Name ?? "lambda_method";
+                sb.AppendLine("object " + name + "(object[])");
+                sb.AppendLine("{");
+
+                sb.AppendLine("  .locals " + _interpreter.LocalCount);
+                sb.AppendLine("  .maxstack " + _interpreter.Instructions.MaxStackDepth);
+                sb.AppendLine("  .maxcontinuation " + _interpreter.Instructions.MaxContinuationDepth);
```

Comment:
Do we care about performance/allocations/etc. in this code?  If so, rather than lines like:
```C#
sb.Append("  .locals " + _interpreter.LocalCount);
```
it's likely better to do:
```C#
sb.Append("  .locals ").Append(_interpreter.LocalCount).AppendLine();
```
so as to avoid an unnecessary box and intermediate string allocation.